units/patch_intent.go

```diff
@@ -522,14 +523,27 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 	}
 
 	patchDoc.BuildVariants = previousPatch.BuildVariants
-
 	if failedOnly {
 		if err = setTasksToPreviousFailed(patchDoc, previousPatch, project); err != nil {
 			return "", errors.Wrap(err, "settings tasks to previous failed")
 		}
-
 	} else {
-		patchDoc.Tasks = previousPatch.Tasks
+		// only add activated tasks from previous patch
+		query := db.Query(bson.M{
+			task.ProjectKey:     project.Identifier,
+			task.VersionKey:     previousPatch.Version,
+			task.DisplayNameKey: bson.M{"$in": previousPatch.Tasks},
```

Comment:
Want to check on the index question; does using display name get us to a better index? Otherwise it seems redundant since any tasks matching the version should be considered

Reply:
@bynn approval actually pending this; can you confirm that you've verified there's a good index fit for this?

Reply:
Idk what you're talking about 👀